server/routes/configRoutes.ts

```diff
@@ -244,7 +244,11 @@ export function configRoutes(router: IRouter, dataSourceEnabled: boolean) {
   router.get(
     {
       path: NODE_API.GET_AVAILABLE_FEATURES,
-      validate: false,
+      validate: dataSourceEnabled ? {
+        query: schema.object({
+          dataSourceId: schema.string(),
```

Comment:
Should this be schema.maybe(schema.string())? If MDS enabled, will we always have dataSourceId in the query?